.github/workflows/ci.yml

```diff
@@ -0,0 +1,113 @@
+name: CI
+on: [push, pull_request]
+env:
+  HUGO_VERSION: 0.26
+  SCALA_VERSION: 2.12.10
+jobs:
+  validate:
+    name: Scala ${{ matrix.scala }}, Java ${{ matrix.java }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [8, 11]
+        scala: [2.12.10, 2.13.1]
+    env:
+      SCALA_VERSION: ${{ matrix.scala }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Coursier
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/coursier
+          key: sbt-coursier-cache
+      - name: Cache SBT
+        uses: actions/cache@v1
+        with:
+          path: ~/.sbt
+          key: sbt-${{ hashFiles('**/build.sbt') }}
```

Comment:
Could use help from a Principal YAML Engineer to DRY these steps up.

Reply:
I know YAML supports references and can extend other objects.
Though perhaps putting the yaml into another format and generating it from an actual programming language would be preferable to going that route?
Though, this YAML already looks like mustache-style template variables in it... (a la that kubernetes stuff?)

Reply:
I know YAML has "anchors", which we use in .travis.yml, and I read three articles on them last night and still don't understand them. 
 Job 1 has steps A, B, C, D. Job 2 has steps A, B, C, E. How does one extract A, B, and C?

On a larger scale, I'd reach for Dhall, but we'd have to check in generated source, and it would be a new language and a new dependency for something that just won't change that often. I think pure YAML is sadly the right way here.

Reply:
I made a bunch of anchors, only to find out that they're [not supported](https://github.community/t5/GitHub-Actions/Support-for-YAML-anchors/td-p/30336).